### ISSUE TITLE
Error GL_INVALID_OPERATION in glMapBufferRange

### ISSUE THREAD

**Issue by ElArtista** [gfx-rs/wgpu#2082]
Error:
```
❯ WGPU_BACKEND=GL cargo run --example cube
    Finished dev [unoptimized + debuginfo] target(s) in 0.12s
     Running `target/debug/examples/cube`
[2021-10-16T14:04:20Z WARN  winit::platform_impl::platform::x11::util::randr] XRandR reported that the display's 0mm in size, which is certifiably insane
[2021-10-16T14:04:20Z INFO  winit::platform_impl::platform::x11::window] Guessed window scale factor: 1
[2021-10-16T14:04:20Z INFO  cube::framework] Initializing the surface...
[2021-10-16T14:04:20Z INFO  wgpu_hal::gles::egl] Loading Wayland library to get the current display
[2021-10-16T14:04:20Z INFO  wgpu_hal::gles::egl] Loading X11 library to get the current display
[2021-10-16T14:04:20Z INFO  wgpu_hal::gles::egl] Using X11 platform
[2021-10-16T14:04:20Z INFO  wgpu_hal::gles::egl] Enabling EGL debug output
[2021-10-16T14:04:20Z INFO  wgpu_hal::gles::egl] Display vendor "Mesa Project", version (1, 5)
[2021-10-16T14:04:20Z INFO  wgpu_hal::gles::egl] 	EGL surface: +srgb
[2021-10-16T14:04:20Z INFO  wgpu_hal::gles::egl] 	Trying native-render
[2021-10-16T14:04:20Z INFO  wgpu_hal::gles::egl] 	EGL context: +debug
[2021-10-16T14:04:20Z INFO  wgpu_hal::gles::egl] 	EGL context: +robust access
[2021-10-16T14:04:20Z INFO  wgpu_hal::gles::egl] 	EGL context: +surfaceless
[2021-10-16T14:04:20Z INFO  wgpu_hal::gles::egl] Max label length: 256
[2021-10-16T14:04:20Z INFO  wgpu_hal::gles::egl] Enabling GLES debug output
[2021-10-16T14:04:20Z INFO  wgpu_hal::gles::adapter] Vendor: X.Org
[2021-10-16T14:04:20Z INFO  wgpu_hal::gles::adapter] Renderer: AMD TURKS (DRM 2.50.0 / 5.14.12-arch1-1, LLVM 12.0.1)
[2021-10-16T14:04:20Z INFO  wgpu_hal::gles::adapter] Version: OpenGL ES 3.2 Mesa 21.2.3
[2021-10-16T14:04:20Z INFO  wgpu_hal::gles::adapter] SL version: OpenGL ES GLSL ES 3.20
[2021-10-16T14:04:20Z INFO  wgpu_core::instance] Adapter GL AdapterInfo { name: "AMD TURKS (DRM 2.50.0 / 5.14.12-arch1-1, LLVM 12.0.1)", vendor: 0, device: 0, device_type: DiscreteGpu, backend: Gl }
Using AMD TURKS (DRM 2.50.0 / 5.14.12-arch1-1, LLVM 12.0.1) (Gl)
[2021-10-16T14:04:20Z INFO  wgpu_core::device] configuring surface with SurfaceConfiguration { usage: RENDER_ATTACHMENT, format: Bgra8UnormSrgb, width: 623, height: 699, present_mode: Mailbox }
[2021-10-16T14:04:20Z WARN  wgpu_core::device] Surface does not support present mode: Mailbox, falling back to FIFO
[2021-10-16T14:04:20Z INFO  cube::framework] Initializing the example...
[2021-10-16T14:04:20Z ERROR wgpu_hal::gles::egl] GLES: [API/Error] ID 1 : GL_INVALID_OPERATION in glMapBufferRange(access indicates neither read or write)
```

Reproducible by running either latest wgpu stable or master

**Platform**
```
❯ glxinfo | grep version                                                                                    2m 57s
server glx version string: 1.4
client glx version string: 1.4
GLX version: 1.4
    Max core profile version: 4.3
    Max compat profile version: 4.3
    Max GLES1 profile version: 1.1
    Max GLES[23] profile version: 3.2
OpenGL core profile version string: 4.3 (Core Profile) Mesa 21.2.3
OpenGL core profile shading language version string: 4.30
OpenGL version string: 4.3 (Compatibility Profile) Mesa 21.2.3
OpenGL shading language version string: 4.30
OpenGL ES profile version string: OpenGL ES 3.2 Mesa 21.2.3
OpenGL ES profile shading language version string: OpenGL ES GLSL ES 3.20
```

**Comment by ElArtista** [comment]
Btw, I just tested with latest master, issue seems to be fixed.

**Comment by kvark** [comment]
The fix is included in wgpu-hal-0.11.3
